openlibrary/book_providers.py

```diff
@@ -214,7 +230,7 @@ def get_provider_order(prefer_ia=False) -> list[AbstractBookProvider]:
     return provider_order
 
 
-def get_book_provider(
+def get_book_providers(
     ed_or_solr: Union[Edition, dict]
 ) -> Optional[AbstractBookProvider]:
```

Comment:
This return type should be updated to be a generator.  This is causing pre-commit to fail.

Reply:
```suggestion
) -> Optional[Iterator[AbstractBookProvider]]:
```
I think that something like this will work (@cdrini to confirm/deny).  You'll also have to import `Iterator` from `typing`

Reply:
Oh good find! Not sure, but seems like that should work?

Reply:
Yep! Seems like that's what we want. We don't need the `Optional` anymore https://mypy.readthedocs.io/en/stable/kinds_of_types.html#generators